### PR TITLE
Document @shorthands functions

### PR DESCRIPTION
**Pull request by sswatson**
Docstrings added for the `@shorthands` functions. Some keyword arguments are documented with `_arg_desc` (using the new functions in `utils`), and some of those descriptions are updated as well.

### CODE REVIEW

File: src/components.jl [hunk 1]
```diff
@@ -256,8 +256,28 @@ mutable struct Font
   color::Colorant
 end
 
-"Create a Font from a list of unordered features"
-function font(args...)
+"""
```
Comment (mkborregaard) [inline src/components.jl]: I like this suggestion for font and text, but I think it is separate from the documentation - would you mind splitting this out?

**Comment by mkborregaard** [comment]
This mostly addresses #1157

File: src/components.jl (hunk 1 above)
Comment (sswatson) [reply src/components.jl]: Yes, this was a mistake. Should be fixed in the new commit.

File: src/shorthands.jl [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+"""
+
+"""
+    scatter(x,y)
+    scatter!(x,y)
+
+Make a scatter plot of y vs x. 
+
+# Arguments 
+$xydoc
```
[the remark this answers is not in the document]
Comment (sswatson) [reply src/shorthands.jl]: That's fair. I don't feel strongly about this either way.

**Comment by sswatson** [comment]
Thanks! Thanks for bearing with me; I've had to figure out a few things with the PR process.

File: src/shorthands.jl [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+Draw contour lines of the `Surface` z. 
+
+# Arguments
+$xyzdoc
+- `levels`: Contour levels (if `AbstractVector`) or number of levels (if `Integer`)
+- `fill`: Bool. Fill area between contours or draw contours only (false by default)
+
+# Example 
+```julia-repl
+julia> contour(-20:20,-20:20,(x,y)->x^2+y^2)
```
Comment (iblis17) [inline src/shorthands.jl]:
```suggestion
x = y = range(-20, 20, length = 100)
julia> contour(x, y, (x, y) -> x^2 + y^2)
```

File: src/shorthands.jl [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+    surface!(x,y,z)
+
+Draw a 3D surface plot. 
+
+# Arguments 
+$xyzdoc
+
+# Example
+```julia-repl 
+julia> surface(1:10,1:10,randn(10,10))
```
Comment (iblis17) [inline src/shorthands.jl]:
```suggestion
julia> x = y = range(-3, 3, length = 100)
julia> surface(x, y, (x, y) -> sinc(norm([x, y])))
```

**Comment by daschw** [comment]
LGTM IMHO, or are there any unresolved issues @mkborregaard ?

File: src/shorthands.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkborregaard) [reply src/shorthands.jl]: Could you take it out?

[the remark this answers is not in the document]
Comment (mkborregaard) [reply src/shorthands.jl]: The reason being that Plots really tries to not be too fussed about input valuse

**Comment by mkborregaard** [comment]
I think personally that the xydoc and xyzdoc are superfluous. 

Other than that this is ready to merge. And a really, really nice PR!

**Comment by mkborregaard** [comment]
@sswatson would you open PRs on the two other functionalities you took out (font keywords and font rotation)? I'm pretty sure we want the font rotation and would be  good to discuss the font keywords (+ the documentation here actually assumes it)

File: src/utils.jl [hunk 5]
```diff
@@ -1199,3 +1199,40 @@ end
...
+        end
+    end
+end
+
+function _document_argument(S::AbstractString)
+    _fmt_paragraph("`$S`: "*_arg_desc[Symbol(S)],leadingspaces = 6 + length(S))
+end
+
+
+
```
Comment (iblis17) [inline src/utils.jl]: superfluous blank lines here.

Comment (sswatson) [reply src/utils.jl]: Fixed, thx.

**Comment by mkborregaard** [comment]
Thanks! Sorry to be particular, but, merging master into your branch isn't actually rebasing - I'll just do a rebase-and-merge, though. Thanks a lot!

File: src/shorthands.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (sswatson) [reply src/shorthands.jl]: Sure, done.

**Comment by daschw** [comment]
Thanks a lot for this @sswatson !

**Pull request merged**